src/app/index/metadata.rs

```diff
@@ -48,39 +49,22 @@ pub fn read(path: &Path) -> Option<SongTags> {
 	}
 }
 
-fn read_id3(path: &Path) -> Result<SongTags> {
-	let tag = {
-		match id3::Tag::read_from_path(&path) {
-			Ok(t) => Ok(t),
-			Err(e) => {
-				if let Some(t) = e.partial_tag {
-					Ok(t)
-				} else {
-					Err(e)
-				}
-			}
-		}?
-	};
-	let duration = {
-		mp3_duration::from_path(&path)
-			.map(|d| d.as_secs() as u32)
-			.ok()
-	};
-
+fn read_id3(tag: &id3::Tag) -> SongTags {
```

Comment:
Now that this function simply performs the conversion between similar tag types, I think it would be nicer to implement it as a `From` trait on `SongTags`.

Reply:
Sure!